src/test/java/io/aiven/kafka/connect/s3/config/S3SinkConfigTest.java

```diff
@@ -690,4 +689,146 @@ void notSupportYyyyUppercaseInFilenameTemplate() {
+        props.put(S3SinkConfig.AWS_S3_REGION_CONFIG, Regions.US_EAST_1.getName());
+
+        final var conf = new S3SinkConfig(props);
+
+        assertEquals("arn:aws:iam::12345678910:role/S3SinkTask", conf.getStsRole().getArn());
+        assertEquals("EXTERNAL_ID", conf.getStsRole().getExternalId());
+        assertEquals("SESSION_NAME", conf.getStsRole().getSessionName());
+        assertEquals(Regions.US_EAST_1, conf.getAwsS3Region());
+    }
+
+    @Test
+    void stsRoleEmptyArn() {
+        final var props = new HashMap<String, String>();
+
+        props.put(S3SinkConfig.AWS_STS_ROLE_EXTERNAL_ID, "EXTERNAL_ID");
+        props.put(S3SinkConfig.AWS_STS_ROLE_SESSION_NAME, "SESSION_NAME");
+        props.put(S3SinkConfig.AWS_S3_BUCKET_NAME_CONFIG, "THE_BUCKET");
+        props.put(S3SinkConfig.AWS_S3_REGION_CONFIG, Regions.US_EAST_1.getName());
+
+        final Throwable t = assertThrows(
+            ConfigException.class,
+            () -> new S3SinkConfig(props)
+        );
+        assertEquals(
+                "Either {aws.access.key.id, aws.secret.access.key} or"
+                        + " {aws_access_key_id, aws_secret_access_key} or"
+                        + " {aws.sts.role.arn, aws.sts.role.external.id, aws.sts.role.session.name} should be set",
+                t.getMessage()
+        );
+    }
+
+    @Test
+    void stsRoleEmptyExternalId() {
+        final var props = new HashMap<String, String>();
+
+        props.put(S3SinkConfig.AWS_STS_ROLE_ARN, "arn:aws:iam::12345678910:role/S3SinkTask");
+        props.put(S3SinkConfig.AWS_STS_ROLE_SESSION_NAME, "SESSION_NAME");
+        props.put(S3SinkConfig.AWS_S3_BUCKET_NAME_CONFIG, "THE_BUCKET");
+        props.put(S3SinkConfig.AWS_S3_REGION_CONFIG, Regions.US_EAST_1.getName());
+
+        final Throwable t = assertThrows(
+            ConfigException.class,
+            () -> new S3SinkConfig(props)
+        );
+        assertEquals(
+                "Either {aws.access.key.id, aws.secret.access.key} or"
+                        + " {aws_access_key_id, aws_secret_access_key} or"
+                        + " {aws.sts.role.arn, aws.sts.role.external.id, aws.sts.role.session.name} should be set",
+                t.getMessage()
+        );
+    }
+
+    @Test
+    void stsRoleEmptySessionName() {
+        final var props = new HashMap<String, String>();
+
+        props.put(S3SinkConfig.AWS_STS_ROLE_ARN, "arn:aws:iam::12345678910:role/S3SinkTask");
+        props.put(S3SinkConfig.AWS_STS_ROLE_EXTERNAL_ID, "EXTERNAL_ID");
+        props.put(S3SinkConfig.AWS_S3_BUCKET_NAME_CONFIG, "THE_BUCKET");
+        props.put(S3SinkConfig.AWS_S3_REGION_CONFIG, Regions.US_EAST_1.getName());
+
+        final Throwable t = assertThrows(
+            ConfigException.class,
+            () -> new S3SinkConfig(props)
+        );
+        assertEquals(
+                "Either {aws.access.key.id, aws.secret.access.key} or"
+                        + " {aws_access_key_id, aws_secret_access_key} or"
```

Comment:
We're getting rid of config options with `_`, let's tell about them in the help strings.

Reply:
But they are still supported, right?

Reply:
I remove _ options from help Strings and added a logging `Error` if deprecated option is used.

Reply:
Yep, they are supported until we find a way how to most transparently get rid of them.

> I remove _ options from help Strings and added a logging Error if deprecated option is used.
:+1: